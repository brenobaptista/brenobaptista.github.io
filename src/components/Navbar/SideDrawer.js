import React from 'react'
import PropTypes from 'prop-types'

import Logo from './Logo'
import NavigationItems from './NavigationItems'
import Backdrop from './Backdrop'

const SideDrawer = ({ open, closed }) => {
  let attachedClasses = ['sideDrawer', 'close']

  if (open) {
    attachedClasses = ['sideDrawer', 'open']
  }

  return (
    <>
      <Backdrop
        show={open}
        clicked={closed}
      />
      <div className={attachedClasses.join(' ')}>
        <div className='logoSideDrawer'>
          <Logo />
        </div>
        <nav>
          <NavigationItems />
        </nav>
      </div>
    </>
  )
}

SideDrawer.propTypes = {
  open: PropTypes.bool,
  closed: PropTypes.func
}

SideDrawer.defaultProps = {
  open: false,
  closed: () => {}
}

export default SideDrawer
