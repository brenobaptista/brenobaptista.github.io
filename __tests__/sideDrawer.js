import React from 'react'
import { render } from '@testing-library/react'
import SideDrawer from '../src/components/Navbar/SideDrawer'

test('matches snapshot with opened sideDrawer', () => {
  const closed = jest.fn()
  const { asFragment } = render(<SideDrawer open closed={closed} email='' />)

  expect(asFragment(<SideDrawer open closed={closed} email='' />)).toMatchSnapshot()
})

test('matches snapshot with closed sideDrawer', () => {
  const closed = jest.fn()
  const { asFragment } = render(<SideDrawer closed={closed} email='' />)

  expect(asFragment(<SideDrawer closed={closed} email='' />)).toMatchSnapshot()
})
