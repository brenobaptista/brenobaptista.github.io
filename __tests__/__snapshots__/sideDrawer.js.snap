// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with closed sideDrawer 1`] = `
<DocumentFragment>
  <div
    class="sideDrawer close"
  >
    <div
      class="logoSideDrawer"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
    </div>
    <nav>
      <ul
        class="navigationItems"
      >
        <li
          class="navigationItem"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/projects"
          >
            PROJECTS
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/blog"
          >
            BLOG
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with opened sideDrawer 1`] = `
<DocumentFragment>
  <div
    aria-label="backdrop"
    class="backdrop"
    role="button"
    tabindex="0"
  />
  <div
    class="sideDrawer open"
  >
    <div
      class="logoSideDrawer"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
    </div>
    <nav>
      <ul
        class="navigationItems"
      >
        <li
          class="navigationItem"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/projects"
          >
            PROJECTS
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/blog"
          >
            BLOG
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
